vcs/git/patch.go

```diff
@@ -8,6 +8,16 @@ import (
 
 // patch -p0 < diff.patch
 
+func isFileBinaryGit(filename string) (bool, error) {
+	repoRoot, err := repoRoot()
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	out, _ := gitCMD("-C", repoRoot, "diff", "--no-index", "--numstat", "/dev/null", filename)
```

Comment:
Return this error too:

```
out, err := gitCMD("-C", repoRoot, "diff", "--no-index", "--numstat", "/dev/null", filename)
if err != nil {
  return false, errors.Trace(err)
}
```

Reply:
Could you also write a quick comment explaining why this command determines whether a file is binary? It's not immediately obvious to me.

Reply:
Sure, the command is exactly the same as the one for generating the patch, save for the --numstat option which returns statistics about each file in the diff instead of the diff, including whether or the the file is binary or not. It uses the same underlying git function.